src/main/java/com/example/controller/settings/KingSettingsController.java

```diff
@@ -46,7 +46,8 @@ public List<Node> getAllNodes() {
 
     @Override
     public void adjustSettingsConditions(MyGraph<Integer, Integer> graph) {
-        // intentionally do nothing - settings do not depend on graph
+        AlgorithmSetting<Integer> phaseSetting = (AlgorithmSetting<Integer>) algorithmSettings.getSettings().get("phase");
+        phaseSetting.setValue(graph.numVertices());
```

Comment:
Proposed phase should be counted the same way as allowedNumberOfTraitors() in KingAlgorithm